src/platform/backends/shared/linux/backend_utils.cpp

```diff
@@ -86,6 +87,34 @@ auto virtual_switch_subnet(const QString& bridge_name)
     }
     return subnet.toStdString();
 }
+
+void check_min_img_size(const mp::MemorySize& requested_size, const mp::Path& image_path)
+{
+    auto qemuimg_process = mp::ProcessFactory::instance().create_process(
+        std::make_unique<mp::QemuImgProcessSpec>(QStringList{"info", image_path}));
+    auto process_state = qemuimg_process->execute();
+
+    if (!process_state.completed_successfully())
+        throw std::runtime_error(fmt::format("Cannot get image info: qemu-img failed ({}) with output:\n{}",
+                                             process_state.failure_message(),
+                                             qemuimg_process->read_all_standard_error()));
+
+    const auto img_info = QString{qemuimg_process->read_all_standard_output()};
+    const auto capture_name = QStringLiteral("size");
+    const auto pattern = QStringLiteral("^virtual size: .+ \\((?<%1>\\d+) bytes\\)$").arg(capture_name);
+    const auto re = QRegularExpression{pattern, QRegularExpression::MultilineOption};
+
+    if (const auto match = re.match(img_info); match.hasMatch())
+    {
+        const auto min_size = match.captured(capture_name).toStdString();
+        if (requested_size < mp::MemorySize{min_size})
+            throw std::runtime_error(fmt::format("Requested disk ({} bytes) below minimum for this image ({} bytes)",
+                                                 requested_size.in_bytes(), min_size));
```

Comment:
I think we should print this human-readable, we can probably assume GBs here, although now I see `MemorySize` floors, would have to round up for this use case…

I'd be OK with this with a TODO to improve the message later.

Reply:
I considered that, but then users might get `Requested disk (2.2G) below minimum for this image (2.2G)`, because of approximations. I used bytes to be precise and avoid that.

Reply:
So, this would happen when a user wanted to save space:

```
$ multipass launch --disk=1G
Requested disk (1G) below minimum for this image (2.2G)
$ multipass launch --disk=2.2G  # user trying next-best thing
Requested disk (2.2G) below minimum for this image (2.2G)
```

Reply:
Well of course, which is why I mentioned we'd need to `ceil()` the minimum, in which case the minimum would be `2.3G`.

Reply:
I see, I did not get what you meant before. OK then, fine by me. It would be imprecise, but I agree it looks friendlier.

Reply:
`MemorySize` does not support decimals ATM, so I will add the TODO for now.